### ISSUE TITLE
Upgrading Shrine from scala-maven-plugin 3.4.6 to 4.0.0 resulted in a 1.9X increase in build time

### ISSUE THREAD

**Issue by dwalend** [davidB/scala-maven-plugin#365]
I recently upgraded Shrine's scala-maven-plugin from 3.4.6 to 4.2.0. The team noticed a change in build time of about 1.9x.

This may be a duplicate of #364 , with the same "wait for bloop/help with bloop" conclusion . 

I bracketed the problem first to just the change of scala-maven-plugin from 3.4.6 to 4.2.0, then to some difference between from 3.4.6 to 4.0.0 .

With 4.0.0 (or 3.4.6 with <recompileMode>all</recompileMode> ) :

```
[INFO] Reactor Summary for SHRINE 2.1.0-M3-SNAPSHOT:
[INFO] 
[INFO] SHRINE ............................................. SUCCESS [  1.073 s]
[INFO] SHRINE Test Commons ................................ SUCCESS [  4.551 s]
[INFO] SHRINE Utility Code ................................ SUCCESS [ 21.986 s]
... It's full of subprojects! ...
[INFO] Test the http4s https client ....................... SUCCESS [  4.154 s]
[INFO] SHRINE SetUp ....................................... SUCCESS [  0.059 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11:25 min
```

With 3.4.6 and <recompileMode>incremental</recompileMode> :

```
[INFO] Reactor Summary for SHRINE 2.1.0-M3-SNAPSHOT:
[INFO] 
[INFO] SHRINE ............................................. SUCCESS [  3.083 s]
[INFO] SHRINE Test Commons ................................ SUCCESS [  6.795 s]
[INFO] SHRINE Utility Code ................................ SUCCESS [ 18.637 s]
... Still so many subprojects! ...
[INFO] Test the http4s https client ....................... SUCCESS [  1.125 s]
[INFO] SHRINE SetUp ....................................... SUCCESS [  0.087 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  07:09 min
```

If you'd like to see the specifics - by building shrine and maybe taking a lunch break or a long walk:

```
git clone https://open.med.harvard.edu/stash/scm/shrine/shrine.git
cd shrine
mvn clean install
```

Thanks,

David

**Comment by slandelle** [comment]
Incremental compiler, which is now enabled by default, is expected to be slower than non-incremental.
There's nothing to be done about it and I don't expect Bloop to improve anything here.
If speed on a clean compile is a concern for you, don't use the incremental compiler.
Typically, you should most likely be using different settings with maven profiles depending on if you're developing, or compiling on CI.

`mvn clean install with <recompileMode>all</recompileMode>`:

```
[INFO] SHRINE ............................................. SUCCESS [  0.995 s]
[INFO] SHRINE Test Commons ................................ SUCCESS [  3.797 s]
[INFO] SHRINE Utility Code ................................ SUCCESS [ 17.371 s]
```

`mvn clean install with <recompileMode>incremental</recompileMode>`:

```
[INFO] SHRINE ............................................. SUCCESS [  0.813 s]
[INFO] SHRINE Test Commons ................................ SUCCESS [  6.057 s]
[INFO] SHRINE Utility Code ................................ SUCCESS [ 20.896 s]
```

Closing, because there's really nothing we can do here if zinc got slower (probably because of bug fixes) between versions 0.13 and 1.0.

**Comment by slandelle** [comment]
Most likely duplicate of #368
